src/coreclr/vm/jitinterface.cpp

```diff
@@ -9384,6 +9393,422 @@ CorInfoTypeWithMod CEEInfo::getArgType (
     return result;
 }
 
+//Only used for LoongArch64-ABI.
+uint32_t CEEInfo::getFieldTypeByHnd(CORINFO_CLASS_HANDLE cls)
```

Comment:
I believe this should be accompanied by `ThunkInput.txt` related changes from https://github.com/dotnet/runtime/pull/62893 That way you can make code compile in this PR. And all changes to this file (`src/coreclr/vm/jitinterface.cpp`) from same PR can be placed here. At least in this change it would be easier to justify JIT interface changes, rationale would be in the code usage.

Reply:
> I believe this should be accompanied by `ThunkInput.txt` related changes from #62893 That way you can make code compile in this PR. And all changes to this file (`src/coreclr/vm/jitinterface.cpp`) from same PR can be placed here. At least in this change it would be easier to justify JIT interface changes, rationale would be in the code usage.

Thanks,
You are right. I have to update all related files together.

Reply:
Maybe compiling sucessfully by CI is hard, Becasuse there are some related files committed by other PRs.
Especially the ABI's files are scattered.

Reply:
I hear you. I see how ABI related changes, and JIT interface are used here and that. I think it's important to at least make sure that there are no compilation issues at this stage. Do you have any ideas how this (and sibling) code can be sliced in better way for review?

Reply:
> I hear you. I see how ABI related changes, and JIT interface are used here and that. I think it's important to at least make sure that there are no compilation issues at this stage. Do you have any ideas how this (and sibling) code can be sliced in better way for review?

I think adding a new CPU-architecture is different with committing a small patch for PR.
Maybe there is a special way for how to commit and review for adding a new architecture.

Liking the compilation issues, maybe compiling all the patches together is a good way at least for adding a new CPU-architecuture.

But for review the patches, it's better that committing several patches.
That‘s a contradiction.

Reply:
I think that the challenge here is due to large code change that was developed offline for a long period of time. Large deltas like that are always challenging to merge upstream.

This is not the first new architecture port that we got for .NET runtime. The difference is that the previous ports (e.g. Linux x86 or ARM64) were not developed offline. The development for the previous ports was in the upstream repo, that naturally broke them into small manageable PRs.

I would recommend to focus on making forward progress. Break the work into smaller PRs as necessary, get those smaller PRs passing CI, reviewed and merged. Rinse and repeat until the whole port is in.

Reply:
> I think that the challenge here is due to large code change that was developed offline for a long period of time. Large deltas like that are always challenging to merge upstream.
> 
> This is not the first new architecture port that we got for .NET runtime. The difference is that the previous ports (e.g. Linux x86 or ARM64) were not developed offline. The development for the previous ports was in the upstream repo, that naturally broke them into small manageable PRs.
> 
> I would recommend to focus on making forward progress. Break the work into smaller PRs as necessary, get those smaller PRs passing CI, reviewed and merged. Rinse and repeat until the whole port is in.

Yes, now we had done it.
For this PR, I have to ensure this PR compiling ok first, is it right ?

Reply:
Now based on the latest patch, this PR can be compiled sucessfully.
And I will fix all the compiling error within every PRs.